sys/ztimer/Kconfig

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2021 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menuconfig MODULE_ZTIMER
+    bool "ztimer - High level timer abstraction layer"
+    depends on TEST_KCONFIG
+    select MODULE_ZTIMER_CORE
+    select MODULE_ZTIMER_CONVERT_FRAC
+    select MODULE_ZTIMER_CONVERT_SHIFT
+    select MODULE_ZTIMER_EXTEND
+
+if MODULE_ZTIMER
+
+menu "Backends"
+
+config MODULE_ZTIMER_PERIPH_RTC
+    bool "RTC peripheral"
+    depends on HAS_PERIPH_RTC
+    select MODULE_PERIPH_RTC
+    select HAVE_ZTIMER_BACKEND
+
+config MODULE_ZTIMER_PERIPH_RTT
+    bool "RTT peripheral"
+    depends on HAS_PERIPH_RTT
+    select MODULE_PERIPH_RTT
+    select HAVE_ZTIMER_BACKEND
+
+config MODULE_ZTIMER_PERIPH_TIMER
+    bool "Timer peripheral"
+    depends on HAS_PERIPH_TIMER
+    select MODULE_PERIPH_TIMER
+    select HAVE_ZTIMER_BACKEND
+
+config MODULE_ZTIMER_MOCK
+    bool "Mock (for testing)"
+    select HAVE_ZTIMER_BACKEND
+    help
+        This ztimer module implements a virtual clock that can be used for
+        unittests. It can be manually adjusted to different timestamps and
+        manually fired to simulate different scenarios and test the ztimer
+        implementation using this as a backing timer.
+
+config HAVE_ZTIMER_BACKEND
+    bool
+    help
+        Indicates that a ztimer backend is provided.
+
+endmenu # Backends
+
+menu "Clocks"
+
+config MODULE_ZTIMER_USEC
+    bool "Microseconds"
+    depends on MODULE_ZTIMER_PERIPH_TIMER
+
+config MODULE_ZTIMER_MSEC
+    bool "Milliseconds"
+    depends on HAVE_ZTIMER_BACKEND
+
+endmenu # Clocks
+
+
+menu "Frequency conversion"
+
+config MODULE_ZTIMER_CONVERT_MULDIV64
+    bool "64-bits arithmetic conversion"
+    select MODULE_ZTIMER_CONVERT
+
+config MODULE_ZTIMER_CONVERT_FRAC
+    bool "Fractional conversion"
+    select MODULE_ZTIMER_CONVERT
+    select MODULE_FRAC
+
+config MODULE_ZTIMER_CONVERT_SHIFT
+    bool "Shift conversion"
+    select MODULE_ZTIMER_CONVERT
```

Comment:
Maybe some help would be needed here.

- shift conversion is both fast and super precise, but cannot work for arbitrary frequencies. It's kind of a software prescaler for the underlying clock. So if the hardware clock frequency and the target clock frequency differ by a factor that is a power of two, this is the best choice - otherwise it is simply not usable.
- frac can be used for arbitrary frequency conversions, but trades in precision to gain speed. In cases where shift conversion cannot be used, this is likely the best trade off.
- muldiv64 is very precise, but the overhead is the highest. On MCUs without hardware division this might not be a good choice.

@kaspar030: Care to check above statements?

Reply:
Help added